src/announcer/announcer.js

```diff
@@ -35,15 +36,74 @@ const toggle = (v) => {
 
 const speak = (message, politeness = 'off') => {
   if (active === false) return
-  clearTimeout(debounce)
-  speechSynthesis.cancel()
-  // assertive messages get spoken immediately
-  if (politeness === 'assertive') {
-    speechSynthesis.speak({ value: message })
+
+  return addToQueue(message, politeness)
+}
+
+const pause = (delay) => {
+  if (active === false) return
+
+  return addToQueue(undefined, undefined, delay)
+}
+
+const addToQueue = (message, politeness, delay = false) => {
+  // keep track of the id so message can be canceled
+  const id = count++
+
+  // setup a promise to allow developer to chain functionality
+  // when specific utterances are done
+  let resolveFn
+  const done = new Promise((resolve) => {
+    resolveFn = resolve
+  })
+
+  // augment the promise with a cancel function
+  done.cancel = () => {
+    const index = queue.findIndex((item) => item.id === id)
+    if (index !== -1) queue.splice(index, 1)
+    resolveFn('canceled')
+  }
+
+  // add message of pause
+  if (delay == false) {
+    politeness === 'assertive'
+      ? queue.unshift({ message, resolveFn, id })
+      : queue.push({ message, resolveFn, id })
   } else {
-    debounce = setTimeout(() => {
-      speechSynthesis.speak({ value: message })
-    }, 400)
+    queue.push({ delay, resolveFn, id })
+  }
+
+  setTimeout(() => {
+    processQueue()
+  }, 100)
+
+  return done
+}
+
+const processQueue = async () => {
+  if (isProcessing === true || queue.length === 0) return
+  isProcessing = true
+
+  const { message, resolveFn, delay } = queue.shift()
+
+  if (delay) {
+    setTimeout(() => {
+      isProcessing = false
+      resolveFn('finished')
+      processQueue()
+    }, delay)
```

Comment:
if delay early return, drop the else (less branching)

of beter helemaal als delay minder common is, dan if (!delay) hot path on top voor betere predictability